openwpm_utils/blocklist.py

```diff
@@ -1,29 +1,110 @@
+# To 
+# https://help.eyeo.com/en/adblockplus/how-to-write-filters#options
+# With help from
+# https://github.com/gorhill/uBlock/blob/1d5800629aaca0d152127d844ca7f6cf975f2f68/src/js/static-net-filtering.js#L55
+# https://github.com/gorhill/uBlock/blob/1d5800629aaca0d152127d844ca7f6cf975f2f68/src/js/static-net-filtering.js#L107
+# Import both objects and use
+# let directMapping = {}
+# Object.entries(typeNameToTypeValue).forEach(k => {directMapping[k[0]]= typeValueToTypeName[k[1] >>4]})
+# to generate an object that represents the direct mapping between these types
+
+type_to_option = {
+"beacon":"ping",
+"csp_report":"other",
+"font":"font",
+"image":"image",
+"imageset":"image",
+"main_frame":"document",
+"media":"media",
+"object":"object",
+"object_subrequest":"object",
+"ping":"ping",
+"script":"script",
+"speculative":"other",
+"stylesheet":"stylesheet",
+"sub_frame":"subdocument",
+"web_manifest":"other",
+"websocket":"websocket",
+"xbl":"other",
+"xml_dtd":"other",
+"xmlhttprequest":"xmlhttprequest",
+"xslt":"other",
+"other":"other"
+}
+
+def get_option_dict(url, top_level_url, resource_type=None):
+    """Build an options dict for BlockListParser.
+    
+    These options are checked here:
+    * https://github.com/englehardt/abp-blocklist-parser/blob/40f6bb5b91ea403b7b9852a16d6c57d5ec26cf7f/abp_blocklist_parser/RegexParser.py#L104-L117
+    * https://github.com/englehardt/abp-blocklist-parser/blob/40f6bb5b91ea403b7b9852a16d6c57d5ec26cf7f/abp_blocklist_parser/RegexParser.py#L240-L248
 
     Parameters
     ----------
     url : string
         The URL of the requested resource.
     top_level_url : string
         The URL of the top-level frame of the requested resource
-    content_type : int
-        An integer representing the content type of the load. Mapping given in:
-        https://searchfox.org/mozilla-central/source/dom/base/nsIContentPolicy.idl
+    resource_type : string
+        All possible values are here https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/ResourceType
 
     Returns
     -------
     dict
         An "options" dictionary for use with BlockListParser
     """
     options = {}
-    options["image"] = content_type == 3
-    options["script"] = content_type == 2
+    # Add type option. Value doesn't matter.
+    if resource_type:
+        try:
+            options[type_to_option[resource_type]] = True
+        except KeyError:
+            raise ValueError(
+                "Argument %s given for `resource_type` not found in map." % resource_type
+            )
     options["domain"] = urlparse(top_level_url).hostname
-    options["third-party"] = get_ps_plus_1(url) != get_ps_plus_1(top_level_url)
+    
+    # Add third-party option if third party. Value doesn't matter.
+    if du.get_ps_plus_1(url) != du.get_ps_plus_1(top_level_url):
+        options["third-party"] = True
     return options
+
+def get_matching_rules(url, top_level_url, resource_type):
```

Comment:
I think we should implement this function to operate on a single blocker object. Thus it will need to make an options dict and a blocker as arguments.